File: Solutionsin Haskell/p020.hs
{- 
 - Solution to Project Euler problem 20
 - Copyright (c) Sankha Saha (CODR). All rights reserved.
 - 
 - https://github.com/codr07/Project-Euler-Solusions
 -}

import qualified EulerLib


{- 
 - We do a straightforward product thanks to Haskell's built-in arbitrary precision Integer type.
 -}

main = putStrLn (show ans)
ans = EulerLib.digitSum (EulerLib.factorial 100)

File: Solutions in Python/p020.py
# 
# Solution to Project Euler problem 20
# Copyright (c) Sankha Saha (CODR). All rights reserved.
# 
#
# https://github.com/codr07/Project-Euler-Solusions
# 

import math


# We do a straightforward computation thanks to Python's built-in arbitrary precision integer type.
def compute():
	n = math.factorial(100)
	ans = sum(int(c) for c in str(n))
	return str(ans)


if __name__ == "__main__":
	print(compute())
